src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1195,6 +1223,100 @@ def construct_payload(self):
         super().construct_payload()
         self.set_up_service_binds()
         self.set_up_extended_location()
+        self.set_up_source()
+        self.set_up_repo()
+
+    def set_up_source(self):
+        if self.get_argument_source():
+            self._set_up_source_or_repo()
```

Comment:
I try to review the `_set_up_source_or_repo`, but it is a little difficult to understand the code logic.
Can we separate logic for `if self.get_argument_source()` and `if self.get_argument_repo()` in `_set_up_source_or_repo`?
It would be more clearly to review what action done in `construct_payload()` and `post_process()`.

Reply:
`--repo` and `--source` re-use a lot of overlapping code from the `-set_up_source_or_repo` method, so it would be redundant to separate them in different methods.

I can explain how `--source` and `--repo` works a bit more here, hopefully that helps with the reviews.

1. If cx used `--source` with `az containerrapp create` a new image is built and pushed to the container registry before the container app is created in `construct_payload()`
This is why we are updating `self.containerapp_def["properties"]["template"]["containers"][0]["image"]` to the new image that was built and pushed by using `--source` [here](https://github.com/snehapar9/azure-cli-extensions/blob/snehapar/Add-source-to-containerapp-update/src/containerapp/azext_containerapp/containerapp_decorator.py#L1310) 
2. If cx used `--repo` with `az containerapp create` the app in the repo is built and deployed by source control. It is executed as a part of `post_process()` because source control requires a container app to be pre-existing in order to be set up. It cannot be executed as a part of `contruct_payload()` because the container app would have not been created.

Reply:
OK, I see.
The redundant is code for construct `app `and `env`, maybe can move this part of code to a function such as: `_construct_app_and_env_for_source_or_repo(self)`, and return `app, env`.

Reply:
Sounds good. Updated.